@here/harp-mapview/lib/VisibleTileSet.ts

```diff
@@ -831,107 +886,141 @@ export class VisibleTileSet {
         }
     }
 
-    // Computes the visible tiles for each supplied datasource.
-    private getVisibleTilesForDataSources(
-        worldCenter: THREE.Vector3,
+    private getVisibleTilesForDataSource(
+        dataSource: DataSource,
         zoomLevel: number,
-        dataSources: DataSource[],
+        rootTiles: TileKeyEntry[],
+        frustumIntersection: FrustumIntersection,
         elevationRangeSource: ElevationRangeSource | undefined
-    ): {
-        tiles: Array<{ dataSource: DataSource; visibleTiles: TileKeyEntry[] }>;
-        allBoundingBoxesFinal: boolean;
-    } {
-        const tiles = [];
-        let allBoundingBoxesFinal: boolean = true;
-
-        for (const dataSource of dataSources) {
-            const displayZoomLevel = dataSource.getDisplayZoomLevel(zoomLevel);
-
-            const tilingScheme = dataSource.getTilingScheme();
-            const useElevationRangeSource: boolean =
-                elevationRangeSource !== undefined &&
-                elevationRangeSource.getTilingScheme() === tilingScheme;
-
-            const tileBounds = new THREE.Box3();
-            const workList: TileKeyEntry[] = this.getRequiredInitialRootTileKeys(worldCenter);
+    ): { visibleTiles: TileKeyEntry[]; bboxesFinal: boolean } {
```

Comment:
My general comment (with one about caches) is that this function should be idempotent and just return brand new `FrustumIntersection` basing on `tilingScheme`, `rootTiles`, `elevationRangeSource` and `maxZoomLevel`.

So, it should be named `getFrustumIntersection`.

This would make it perfectly compatible with `cache` concept and would make `FrustumInterfsectionCache` class almost redundant as it would be just plain `Map<TilingScheme, FrustumIntersection>` local to `getVisibleTilesForDataSources`.

Note on `maxZoomLevel` in next comment.

Reply:
done.